firebase-firestore/src/main/java/com/google/firebase/firestore/model/TargetIndexMatcher.java

```diff
@@ -186,6 +186,55 @@ public boolean servedByIndex(FieldIndex index) {
     return true;
   }
 
+  /**
+   * Returns whether the given index is a partial index for the current target.
+   *
+   * <p>A partial index is one which does not have a segment for every field filter and every
+   * orderBy in the query.
+   */
+  public boolean isPartialIndex(FieldIndex index) {
+    List<FieldFilter> allFilters = new ArrayList<>(equalityFilters);
```

Comment:
It looks like the answer is "no" since our explicit orderBy only included the inequality segments, and not all equality segments: https://osscs.corp.google.com/firebase-sdk/firebase-android-sdk/+/master:firebase-firestore/src/main/java/com/google/firebase/firestore/core/Query.java;l=341